allensdk/brain_observatory/behavior/dprime.py

```diff
@@ -0,0 +1,103 @@
+    go_responses = np.empty_like(hit, dtype=np.float)
+    go_responses.fill(float('nan'))
+    go_responses[hit] = 1
+    go_responses[miss] = 0
+    return go_responses
+
+
+def get_hit_rate(hit=None, miss=None, aborted=None, sliding_window=100):
+    go_responses = get_go_responses(hit=hit, miss=miss, aborted=aborted)
+    hit_rate = pd.Series(go_responses).rolling(window=sliding_window, min_periods=0).mean().values
+    return hit_rate
+
+
+def get_trial_count_corrected_hit_rate(hit=None, miss=None, aborted=None, sliding_window=100):
+    go_responses = get_go_responses(hit=hit, miss=miss, aborted=aborted)
+    go_responses_count = pd.Series(go_responses).rolling(window=sliding_window, min_periods=0).count()
+    hit_rate = pd.Series(go_responses).rolling(window=sliding_window, min_periods=0).mean().values
+    trial_count_corrected_hit_rate = np.vectorize(trial_number_limit)(hit_rate, go_responses_count)
+    return trial_count_corrected_hit_rate
+
+
+def get_catch_responses(correct_reject=None, false_alarm=None, aborted=None):
+    assert len(correct_reject) == len(false_alarm) == len(aborted)
+    not_aborted = np.logical_not(np.array(aborted, dtype=np.bool))
+    correct_reject = np.array(correct_reject, dtype=np.bool)[not_aborted]
+    false_alarm = np.array(false_alarm, dtype=np.bool)[not_aborted]
+
+    # Catch responses are nan when go (aborted are masked out); 0 for correct-rejection, 1 for false-alarm
+    # This allows pd.Series.rolling to ignore non-catch trial data
+    catch_responses = np.empty_like(correct_reject, dtype=np.float)
+    catch_responses.fill(float('nan'))
+    catch_responses[false_alarm] = 1
+    catch_responses[correct_reject] = 0
+    return catch_responses
+
+
+def get_false_alarm_rate(correct_reject=None, false_alarm=None, aborted=None, sliding_window=100):
+    catch_responses = get_catch_responses(correct_reject=correct_reject, false_alarm=false_alarm, aborted=aborted)
+    false_alarm_rate = pd.Series(catch_responses).rolling(window=sliding_window, min_periods=0).mean().values
+    return false_alarm_rate
+
+
+def get_trial_count_corrected_false_alarm_rate(correct_reject=None, false_alarm=None, aborted=None, sliding_window=100):
+    catch_responses = get_catch_responses(correct_reject=correct_reject, false_alarm=false_alarm, aborted=aborted)
+    catch_responses_count = pd.Series(catch_responses).rolling(window=sliding_window, min_periods=0).count()
+    false_alarm_rate = pd.Series(catch_responses).rolling(window=sliding_window, min_periods=0).mean().values
+    trial_count_corrected_false_alarm_rate = np.vectorize(trial_number_limit)(false_alarm_rate, catch_responses_count)
+    return trial_count_corrected_false_alarm_rate
+
+
+def get_rolling_dprime(rolling_hit_rate, rolling_fa_rate, limits=(0.01, 0.99)):
+    return np.array([get_dprime(hr, far, limits=limits) for hr, far in zip(rolling_hit_rate, rolling_fa_rate)])
+
+
+def get_dprime(hit_rate, fa_rate, limits=(0.01, 0.99)):
```

Comment:
The default has been hardcoded with a default window size of 100. It would be nice to couple this to the rolling window size.